### PR TITLE
Fix tsp, update cadl-ranch 0.31.8

### PR DESCRIPTION
**Pull request by v-hongli1**
Fixed tsp, update cadl-ranch 0.31.8

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
your action run? result?

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
merge it if CI pass

File: typespec-tests/src/test/java/com/type/property/additionalproperties/ExtendsTests.java [hunk 1]
```diff
@@ -65,13 +65,16 @@ public void testExtendsModelArrayClient() {
     public void testExtendsModelClient() {
         ExtendsModelAdditionalProperties body = new ExtendsModelAdditionalProperties();
         Map<String, ModelForRecord> propertyMap = new LinkedHashMap<>();
+        propertyMap.put("knownProp", new ModelForRecord("ok"));
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/type/property/additionalproperties/ExtendsTests.java:68]: "knownProp" should not be set via additionalPorperties, and should not be retrieved from additionalPorperties

If it cannot be set/get via model, disable the test, mark it as a bug.

Comment (v-hongli1) [reply typespec-tests/src/test/java/com/type/property/additionalproperties/ExtendsTests.java]: Disabled this test case, and `IsModelArrayClientTest#testPullAndGet` and `IsModelClientTest#testPullAndGet` same as this.

File: typespec-tests/src/test/java/com/type/property/additionalproperties/IsModelClientTest.java [hunk 2]
```diff
@@ -3,25 +3,20 @@
...
     public void testPullAndGet() {
-        Map<String, ModelForRecord> properties = new LinkedHashMap<>();
-        properties.put("prop", new ModelForRecord("ok"));
-        client.put(properties);
-
-        properties = client.get();
-        Assertions.assertNotNull(properties);
-        Assertions.assertNotNull(properties.get("prop"));
-        Assertions.assertEquals("ok", properties.get("prop").getState());
+//        Map<String, ModelForRecord> properties = new LinkedHashMap<>();
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/type/property/additionalproperties/IsModelClientTest.java:13]: Why comment out?

Comment (v-hongli1) [reply typespec-tests/src/test/java/com/type/property/additionalproperties/IsModelClientTest.java:13]: Please following as https://github.com/Azure/autorest.java/pull/2692/files#diff-ce230670765308677adff8ade0554d3d5f2935e4ad02368843af75c2150d2875R127

**Review by weidongxu-microsoft — APPROVED** [review]

File: typespec-tests/src/test/java/com/type/property/additionalproperties/IsModelClientTest.java (hunk 2 above)
Comment (weidongxu-microsoft) [reply typespec-tests/src/test/java/com/type/property/additionalproperties/IsModelClientTest.java]: OK. So no test disabled or commented?

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**